src/util/NuulsUploader.cpp

```diff
@@ -0,0 +1,216 @@
+#include "NuulsUploader.hpp"
+
+#include "common/Env.hpp"
+#include "common/NetworkRequest.hpp"
+#include "providers/twitch/TwitchMessageBuilder.hpp"
+
+#include <QBuffer>
+#include <QHttpMultiPart>
+#include <QMimeDatabase>
+#include <QMutex>
+
+#define UPLOAD_DELAY 2000
+// Delay between uploads in milliseconds
+
+namespace {
+
+boost::optional<QByteArray> convertToPng(QImage image)
+{
+    QByteArray imageData;
+    QBuffer buf(&imageData);
+    buf.open(QIODevice::WriteOnly);
+    bool success = image.save(&buf, "png");
+    if (success)
+    {
+        return boost::optional<QByteArray>(imageData);
+    }
+    else
+    {
+        return boost::optional<QByteArray>(boost::none);
+    }
+}
+}  // namespace
+
+namespace chatterino {
+// These variables are only used from the main thread.
+auto uploadMutex = QMutex();
+std::queue<RawImageData> uploadQueue;
```

Comment:
might cause violations to one definition rule if the same name is defined in different cpp files

Reply:
It should be static I think

Reply:
To get internal linkage

Reply:
Made both static in 8f076d8